tools/extensions/validate_extensions.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+
+# Validate extension metadata
+
+# This script expects a copy of the envoy source to be located at /source
+# Alternatively, you can specify a path to the source dir with `ENVOY_SRCDIR`
```

Comment:
I think it should default to using `$PWD` and then in Docker set `/source` via `ENVOY_SRCDIR`. The only place we expect to see `/source` is in Docker.

Reply:
this is very temporary - in the next PR most of this will be gone

Reply:
also - re "$PWD" - this cant do that because its inside the bazel build dir - a path must be passed, there cant be a default that is not hardcoded.

this is one of the reasons for doing this - we wont need the envoy_srcdir to be set at all once this code is trimmed down - the bazel env should have anything that is required

Reply:
also worth mentioning this is just a repurposing of the previous `generate_extension_db` script - hence the inclusion of the `/source` default etc

Reply:
Why do we need to merge this specific part of the PR if it goes in the next one?

Reply:
its just maintaining the equivalence of `generate_extension_db` - ie ensuring it works in the same circumstances (of being called from `docs/build.sh`